### PR TITLE
feat(alerts): tokens implementation

### PR DESCRIPTION
**Pull request by mindarelus**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: tests/snapshots/alert/alerts--variants-core-dark.png
Comment (Jinnie) [inline tests/snapshots/alert/alerts--variants-core-dark.png:1]: The black text on dark green in the fourth example carousel looks very suspicious from a11y perspective.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 1]
```diff
@@ -197,20 +206,24 @@
   //close alert button.
   .alert-item > span,
   .alert-text {
-    display: inline-block; //needed for IE11
-    flex-grow: 1;
+    // display: inline-block; //needed for IE11
```
Comment (Jinnie) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:209]: remove commented lines here and the 2 that are couple of line below

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 2]
```diff
@@ -376,13 +402,18 @@
       }
     }
 
-    .alert-item > span,
-    .alert-text {
-      flex: 0 1 100%; // IE11 needs 100% here.
-    }
+    // .alert-item > span,
```
Comment (Jinnie) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:405]: remove commented lines

**Review by Jinnie — CHANGES_REQUESTED** [review]

File: tests/snapshots/alert/alerts--variants-core-dark.png
Comment (valentin-mladenov) [reply tests/snapshots/alert/alerts--variants-core-dark.png:1]: Yeah I noticed it too. The property `clr-app-alert-danger-pager-bg-color` is introduced, but not used.
@mindarelus change this line to use the new property. https://github.com/vmware-clarity/ng-clarity/blob/alert-tokens/projects/angular/src/emphasis/alert/_alert.clarity.scss#L588

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 3]
```diff
@@ -428,7 +452,7 @@
 
     .alert-icon-wrapper {
       padding-top: 0;
-      height: $alertSmallLineHeight;
+      height: variables.$clr_baselineRem_0_667;
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:455]: The height must follow 24px icon token.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 4]
```diff
@@ -402,15 +422,19 @@
...
     // sub-pixel required but rounding up to the nearest whole pixel...
     font-size: variables.$clr_baselineRem_0_5 - variables.$clr_baselineRem_1px;
     letter-spacing: normal;
-    line-height: $alertSmallLineHeight;
+    @include mixins.css-var(
+      line-height,
+      clr-alert-sm-line-height,
+      variables.$clr_baselineRem_0_667,
+      variables.$clr-use-custom-properties
+    );
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:437]: We need design for `sm` alerts.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 5]
```diff
@@ -175,42 +175,49 @@
...
-  //close alert button.
   .alert-item > span,
   .alert-text {
-    display: inline-block; //needed for IE11
-    flex-grow: 1;
+    @include mixins.css-var(flex-grow, clr-alert-text-flex-grow, 1, variables.$clr-use-custom-properties);
     flex-shrink: 1;
-    flex-basis: 98%; //needed for IE11
-    max-width: 98%; //needed for IE11
     margin-right: variables.$clr_baselineRem_0_5;
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:208]: Why do we keep this margin in token migration? 
The spacing is done by `gaps`. Keep it only for NG theme.


```suggestion
```

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 6]
```diff
@@ -376,13 +396,13 @@
...
-      margin-top: variables.$clr_baselineRem_0_125;
+      @include mixins.css-var(
+        margin-top,
+        clr-alert-icon-wrapper-margin-top,
+        variables.$clr_baselineRem_0_125,
+        variables.$clr-use-custom-properties
+      );
     }
 
     .close {
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:408]: Close icon height and width must be 16px by design.

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 7]
```diff
@@ -476,7 +505,11 @@
         $smallAlertCloseIconDims: variables.$clr_baselineRem_0_83;
         margin-top: -1 * ($alertSmallNudge + variables.$clr_baselineRem_1px);
         margin-right: -1 * variables.$clr_baselineRem_1px;
-        @include mixins.equilateral($smallAlertCloseIconDims);
+        @include mixins.min-equilateral-var(
+          clr-alert-close-button-dims,
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:509]:
```suggestion
          clr-alert-close-icon-size,
```

File: projects/ui/src/shim.cds-core.scss [hunk 8]
```diff
@@ -455,13 +455,16 @@
...
   --clr-alert-sm-line-height: var(--cds-global-typography-font-size-6);
-  --clr-alert-app-padding: var(--cds-alias-global-space-4);
+  --clr-alert-app-padding: var(--cds-global-space-4);
   --clr-alert-text-flex-grow: 0; // ATM: delete after token migration
   --clr-alert-icon-wrapper-margin-top: 0; // ATM: delete after token migration
-  --clr-alert-items-gap: var(--cds-alias-global-space-5);
+  --clr-alert-items-gap: var(--cds-global-space-5);
+  --clr-alert-items-margin-right: 0; // ATM: delete after token migration
+  --clr-alert-icon-margin-left: 0; // ATM: delete after token migration
+  --clr-alert-close-button-dims: var(--cds-global-space-7);
```
Comment (valentin-mladenov) [inline projects/ui/src/shim.cds-core.scss:467]:
```suggestion
  --clr-alert-close-icon-size: var(--cds-global-space-7);
```

File: projects/angular/src/emphasis/alert/_alert.clarity.scss [hunk 9]
```diff
@@ -326,7 +346,11 @@
       clr-icon {
         $alert-close-icon-dims: variables.$clr_baselineRem_1 - alert-variables.$clr-alert-borderwidth;
         margin-top: -1 * variables.$clr_baselineRem_0_125;
-        @include mixins.equilateral($alert-close-icon-dims);
+        @include mixins.min-equilateral-var(
+          clr-alert-close-button-dims,
```
Comment (valentin-mladenov) [inline projects/angular/src/emphasis/alert/_alert.clarity.scss:350]:
```suggestion
          clr-alert-close-icon-size,
```

**Review by valentin-mladenov — APPROVED** [review]
Check the comments regarding property name change. All other LGTM.

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
Please test or create stories for alerts with actions. It gets broken - check screenshot: 
Before PR changes:
Screenshot 2023-08-16 at 10 26 06
After PR changes:
Screenshot 2023-08-16 at 10 25 44

**Review by dtsanevmw — APPROVED** [review]

**Review by Jinnie — APPROVED** [review]
Thanks for the fixes, looks good to me.

**Pull request merged**